ROOT TBB build failures on GHA

https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/7740032215/job/21104114360?pr=854#step:8:9340
```
CMake Error at /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/builds/ROOT/build/TBB-prefix/src/TBB-stamp/TBB-build-Release.cmake:49 (message):
  Command failed: 2

   'make' 'compiler=gcc' 'cpp0x=1' 'CXXFLAGS= -mno-rtm' 'CPLUS=/usr/lib/ccache/g++-11' 'CONLY=/usr/lib/ccache/gcc-11' 'LDFLAGS='
  See also
    /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/builds/ROOT/build/TBB-prefix/src/TBB-stamp/TBB-build-*.log
```

_Originally posted by @KrisThielemans in https://github.com/SyneRBI/SIRF-SuperBuild/issues/854#issuecomment-1921100780_

we store log files, and I see
```
build/common.inc:48: "Warning: deprecated cpp0x=1 is used, stdver must be used instead. Building in stdver=c++0x mode."
../../build/Makefile.tbb:28: CONFIG: cfg=debug arch=intel64 compiler=gcc target=linux runtime=cc11.4.0_libc2.35_kernel6.2.0
make[7]: *** read jobs pipe: Bad file descriptor.  Stop.
make[7]: *** Waiting for unfinished jobs....
In file included from ../../src/tbb/concurrent_hash_map.cpp:17:
../../include/tbb/concurrent_hash_map.h: In constructor ‘tbb::interface5::internal::hash_map_base::hash_map_base()’:
../../include/tbb/concurrent_hash_map.h:131:24: warning: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘struct tbb::interface5::internal::hash_map_base::bucket’ with no trivial copy-assignment; use value-initialization instead [-Wclass-memaccess]
  131 |             std::memset(my_embedded_segment, 0, sizeof(my_embedded_segment));
      |             ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../include/tbb/concurrent_hash_map.h:93:16: note: ‘struct tbb::interface5::internal::hash_map_base::bucket’ declared here
   93 |         struct bucket : tbb::internal::no_copy {
      |                ^~~~~~
make[6]: *** [Makefile:28: tbb] Error 2
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -17,6 +17,7 @@
   - siemens_to_ismrmrd: v1.2.11
   - CIL: db5a2a6cd3bddfbbf53e65f0549ac206096e5b44 (version of Feb 13, 2024)
   - CCPi-Regularisation-Toolkit: 71f8d304d804b54d378f0ed05539f01aaaf13758 (version of Dec 3. 2023)
+  - ROOT: v6-28-12
 
 ## v3.5.0
 - Temporarily disable CCPi-Regularisation due to #832

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -18,6 +18,7 @@
   - CIL: db5a2a6cd3bddfbbf53e65f0549ac206096e5b44 (version of Feb 13, 2024)
   - CCPi-Regularisation-Toolkit: 71f8d304d804b54d378f0ed05539f01aaaf13758 (version of Dec 3. 2023)
   - ROOT: v6-28-12
+  - nlohmann/json: 3.11.3
 
 ## v3.5.0
 - Temporarily disable CCPi-Regularisation due to #832

--- a/version_config.cmake
+++ b/version_config.cmake
@@ -167,7 +167,7 @@ set(DEFAULT_CCPi-Regularisation-Toolkit_TAG "71f8d304d804b54d378f0ed05539f01aaaf
 
 # CERN ROOT
 set(DEFAULT_ROOT_URL https://github.com/root-project/root)
-set(DEFAULT_ROOT_TAG "v6-26-10")
+set(DEFAULT_ROOT_TAG "v6-28-12")
 
 # ACE
 set(DEFAULT_ACE_URL https://github.com/paskino/libace-conda)

--- a/version_config.cmake
+++ b/version_config.cmake
@@ -155,7 +155,7 @@ set(DEFAULT_SPM_URL https://github.com/spm/SPM12)
 set(DEFAULT_SPM_TAG r7771)
 
 set(DEFAULT_JSON_URL https://github.com/nlohmann/json)
-set(DEFAULT_JSON_TAG v3.10.4)
+set(DEFAULT_JSON_TAG v3.11.3)
 
 # CCPi CIL
 # minimum supported version of CIL supported is > 22.1.0 or from commit a6062410028c9872c5b355be40b96ed1497fed2a

